syntax = "proto3";

package lab6Proto;

message addRequest {
  int32 a = 1;
  int32 b = 2;
}

message addReply {
  int32 c = 1;
}

message rawimageRequest {
  bytes img = 1;
}

message imageReply {
  int32 width = 1;
  int32 height = 2;
}

message dotproductRequest {
  repeated double a = 1;
  repeated double b = 2;
}

message dotproductReply {
  double c = 1;
}

message jsonimageRequest {
  string img = 1;
}

// Interface exported by the server.
service Lab6grpc {
    rpc doAdd(addRequest) returns (addReply) {}
    rpc doDotproduct(dotproductRequest) returns (dotproductReply) {}
    rpc doRawimage(rawimageRequest) returns (imageReply) {}
    rpc doJsonimage(jsonimageRequest) returns (imageReply) {}
}